addon/services/tags.js

```diff
@@ -40,15 +40,16 @@ export default class TagsService extends Service {
   }
 
   /**
-   * Adds a tag to a document and creates the tag if necessary.
+   * Adds a tag to a document and creates the tag if necessary. Returns the added tag
    *
    * @param {Object} document The target document.
    * @param {Object|String} tag Either e tag instance or a name.
+   * @returns {Object} addedTag The added tag
    */
   @action async add(document, tag) {
     if (typeof tag === "string") {
-      const existing = this.allTags.findBy("name", tag);
-
+      tag = tag.trim();
+      const existing = this.allTags.findBy("id", dasherize(tag));
```

Comment:
Nitpick: I think it's bad style to re-assign arguments (doesn't matter for primitive data types, but can cause subtle bugs when args are passed by reference...). I'd declare a new variable for the return value instead of reusing the argument.